src/classes/CLAN_PrimaryLanguage_TEST.cls

```diff
@@ -0,0 +1,268 @@
+        return contactLanguageToInsert;
+    }
+
+    /*******************************************************************************************************
+    * @description Verifies/asserts that the given Language is specified as the Primary Language on the test
+    * Contact.
+    * @param languageId The Language that should be the Primary Language on the Contact. May be null.
+    ********************************************************************************************************/
+    private static void assertPrimaryLanguageOnContact(Id languageId) {
+        List<Contact> queriedContacts = [SELECT Primary_Language__c FROM Contact WHERE Id = :contact.Id];
+        System.assertEquals(1, queriedContacts.size(), 'The expected Contact was not found.');
+        System.assertEquals(languageId, queriedContacts[0].Primary_Language__c, 
+            'The Primary Language on the Contact was not set as expected.');
+    }
+
+    /*******************************************************************************************************
+    * @description Tests that when a Contact Language is created with Primary Language checked, the Primary
+    * Language is set on the associated Contact.
+    ********************************************************************************************************/
+    @isTest
+    private static void setPrimaryLanguageOnInsert() {
+        setup();
+
+        assertPrimaryLanguageOnContact(null);
+
+        Test.startTest();
+        insertContactLanguage(language.Id, true);
+        Test.stopTest();
+
+        assertPrimaryLanguageOnContact(language.Id);
+    }
+
+    /*******************************************************************************************************
+    * @description Tests that when a Contact Language is updated with Primary Language checked, the Primary
+    * Language is set on the associated Contact.
+    ********************************************************************************************************/
+    @isTest
+    private static void setPrimaryLanguageOnUpdate() {
+        setup();
+
+        Contact_Language__c contactLanguage = insertContactLanguage(language.Id, false);
+        assertPrimaryLanguageOnContact(null);
+
+        Test.startTest();
+        contactLanguage.Primary_Language__c = true;
+        update contactLanguage;
+        Test.stopTest();
+
+        assertPrimaryLanguageOnContact(language.Id);
+    }
+
+    /*******************************************************************************************************
+    * @description Tests that when multiple Contact Languages are updated with Primary Language checked, the 
+    * Primary Language is set on the associated Contact of the most recent Primary Language and the other
+    * Contact Language record is unset as the Primary Language.
+    ********************************************************************************************************/
+    @isTest
```

Comment:
it might just be because I am tired but I'm having a hard time following this test. Maybe its because the logic is broken out and I have to go back and ensure I am looking at the method you're calling. I'm questioning here though if updating the primary languages at the same time in a list like that guarantees the same order of operations? Can we safely assume that firstLanguage is the one that will get marked as not primary and instead the secondLanguage will be primary? I'm not sure I am making sense.
Generally in unit tests, and maybe we should chat about this -- I actually prefer brittle, explicit code as opposed to re-usable code. (you may feel differently, that's fine). Is this testing creating both at the same time? Or is it testing that one is overwriting the other? (Sorry if I am not making sense...)

Reply:
No, your concern is valid and this is the test that I like the least, so thanks for calling it out. The test is updating both at the same time. I don't think we can assume that firstLanguage is the one that will get marked as not primary. Perhaps a better test is to assert that only one of the two records remains Primary after the unit tests runs.

Reply:
Done!